nsqd/http.go

```diff
@@ -114,6 +114,58 @@ func (s *httpServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	s.router.ServeHTTP(w, req)
 }
 
+func (s *httpServer) isAuthEnabled(req *http.Request) bool {
+	if s.isTLS(req) == 1 {
+		return s.ctx.nsqd.IsAuthEnabledHTTPS()
+	}
+	return s.ctx.nsqd.IsAuthEnabledHTTP()
+}
+
+func (s *httpServer) isTLS(req *http.Request) int32 {
+	if req.TLS != nil {
+		return 1
+	}
+	return 0
+}
```

Comment:
I feel like these two methods above might be a little prematurely factored.

1. they're not really re-used
2. `isTLS` doesn't technically need the `s *httpServer` receiver
3. it was initially confusing when seeing _another_ `isAuthEnabled` variant attached to `httpServer`

I vote for just consolidating this into `checkAuth`.  What do you think?